src/main/java/org/opensearch/flowframework/rest/RestCreateWorkflowAction.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.rest;
+
+import com.google.common.collect.ImmutableList;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.flowframework.FlowFrameworkPlugin;
+import org.opensearch.flowframework.model.Template;
+import org.opensearch.flowframework.transport.CreateWorkflowAction;
+import org.opensearch.flowframework.transport.WorkflowRequest;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Rest Action to facilitate requests to create and update a use case template
+ */
+public class RestCreateWorkflowAction extends BaseRestHandler {
+
+    private static final String CREATE_WORKFLOW_ACTION = "create_workflow_action";
+
+    // TODO : move to common values class, pending implementation
+    /**
+     * Field name for workflow Id, the document Id of the indexed use case template
+     */
+    public static final String WORKFLOW_ID = "workflow_id";
+
+    /**
+     * Intantiates a new RestCreateWorkflowAction
+     */
+    public RestCreateWorkflowAction() {
+        // TODO : Pass settings and cluster service to constructor and add settings update consumer for request timeout value
+    }
+
+    @Override
+    public String getName() {
+        return CREATE_WORKFLOW_ACTION;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return ImmutableList.of(
+            // Create new workflow
+            new Route(RestRequest.Method.POST, String.format(Locale.ROOT, "%s", FlowFrameworkPlugin.WORKFLOWS_URI)),
```

Comment:
1. Why do we need a new API to store use case template in GC? Provision API should store the use case template in GC, return a workflow_id and start the provisioning on a different thread.
2. We can just have PUT Request for both the routes. The only difference should be the workflow_id at the end.

Reply:
We've already had this discussion and it was decided to have separate APIs to save a template and to execute a template. This was a requirement of the front end https://github.com/opensearch-project/opensearch-ai-flow-framework/issues/49#issuecomment-1730531046

Reply:
Refer https://github.com/opensearch-project/opensearch-ai-flow-framework/issues/49 for more discussion on the same.